### PR TITLE
Move CSRF check into function to allow override

### PR DESCRIPTION
**Pull request by kyagna**

### CODE REVIEW

File: zuul-core/src/main/java/com/netflix/zuul/netty/server/push/PushAuthHandler.java [hunk 1]
```diff
@@ -73,13 +72,9 @@ protected final void channelRead0(ChannelHandlerContext ctx, FullHttpRequest req
         }
         else if (pushConnectionPath.equals(path)) {
             // CSRF protection
-            final String origin = req.headers().get(HttpHeaderNames.ORIGIN);
-            if (((PushProtocol.WEBSOCKET.getPath().equals(pushConnectionPath))) &&
```
[the remark this answers is not in the document]
Comment (kyagna) [reply zuul-core/src/main/java/com/netflix/zuul/netty/server/push/PushAuthHandler.java:77]: The path returned by PushProtocol.WEBSOCKET.getPath() is passed as pushConnectionPath while constructing the handler. So technically they will always be the same. 
Example init: https://github.com/Netflix/zuul/blob/master/zuul-sample/src/main/java/com/netflix/zuul/sample/push/SampleWebSocketPushChannelInitializer.java#L39

**Review by gavinbunney — APPROVED** [review]

File: zuul-core/src/test/java/com/netflix/zuul/netty/server/push/PushAuthHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        request.headers().add(HttpHeaderNames.ORIGIN, "zuul-push.foo.com");
+        assertTrue(authHandler.isInvalidOrigin(request));
+
+        // Valid input
+        request.headers().remove(HttpHeaderNames.ORIGIN);
+        request.headers().add(HttpHeaderNames.ORIGIN, "zuul-push.netflix.com");
+        assertFalse(authHandler.isInvalidOrigin(request));
+    }
+
+    class ZuulPushAuthHandlerTest extends PushAuthHandler {
```
Comment (argha-c) [inline zuul-core/src/test/java/com/netflix/zuul/netty/server/push/PushAuthHandlerTest.java:50]: I don't think this extension in necessary? You can instantiate the real handler.

File: zuul-core/src/test/java/com/netflix/zuul/netty/server/push/PushAuthHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        ZuulPushAuthHandlerTest authHandler = new ZuulPushAuthHandlerTest();
+
+        final DefaultFullHttpRequest request = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.POST,
+                "/ws", Unpooled.buffer());
+
+        // Invalid input
+        assertTrue(authHandler.isInvalidOrigin(request));
+        request.headers().add(HttpHeaderNames.ORIGIN, "zuul-push.foo.com");
+        assertTrue(authHandler.isInvalidOrigin(request));
+
```
Comment (argha-c) [inline zuul-core/src/test/java/com/netflix/zuul/netty/server/push/PushAuthHandlerTest.java:43]: Valid and invalid origins should be separate tests.

**Comment by zengli00** [comment]
这是来自QQ邮箱的自动回复邮件。
 
您好，已经收到您的邮件，尽快给您回复。                                    --曾礼

**Pull request merged**